eventbus/driver/nats.go

```diff
@@ -202,14 +239,22 @@ func (n *natsStreaming) processEventSourceMsg(m *stan.Msg, msgHolder *eventSourc
 		}
 	}
 
+	// NATS Streaming guarantees At Least Once delivery,
+	// so need to check if the message is duplicate
+	if _, ok := msgHolder.smap.Load(event.ID()); ok {
```

Comment:
@whynowy ok i see. so it makes sense to keep this load and check as a checkpoint if the following lines before load or store are intensive. side note: looking at the dependency handling code it doesn't look thread safe, there's some races for map accesses that should be addressed there

Reply:
Yes, you are right.

Reply:
All right, I think there's no need to take care of that, because the whole function of `processEventSourceMsg` is a blocking call int NATS Streaming client, there's only 1 message being processed at a time. @terev